app/users/admin/edit-user/edit-user.component.ts

```diff
@@ -13,33 +14,56 @@ export class EditUserComponent implements OnInit {
   user: User = User.getEmptyUser();
   showStudentDetails = false;
   roles: string[] = Roles.ROLES;
+  updatedRole: string[] = [];
+  selectedRole: string[];
+  newRole = false;
 
   constructor(private route: ActivatedRoute,
-              private router: Router,
-              private userService: UserService) {}
+    private router: Router,
+    private userService: UserService) { }
 
   ngOnInit(): void {
-    const userId = decodeURI(this.route.snapshot.params['identity']);
+    const userId = decodeURI(this.route.snapshot.params['id']);
     this.userService.getUser(userId)
       .then(
-        (data: User) => {
-          this.user = data;
-          this.showStudentDetails = (this.user.role === 'Student');
-        }
+      (data: User) => {
+        this.user = data;
+        this.showStudentDetails = (this.user.roles === ['Student']);
```

Comment:
Since users can have multiple roles, you should check whether "Student" is one of the user's roles (instead of checking whether the user has *only* the role "Student")